<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Understanding Barcodes</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Understanding Barcodes" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/galaxy-single-cell'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-umis%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/scrna-umis/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Understanding Barcodes",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / scrna-umis / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Understanding Barcodes' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html",
  "timeRequired": "PT2H",
  "keywords": "single-cell",
  "description": "The questions this  addresses are:\n - What are barcodes?\n - What is their purpose?\n - How are barcodes formatted?\n\n\\nThe objectives are:\n - Demultiplex FASTQ data via UMI-tools\n - Understand and validate the extraction of barcodes\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html",
      "name": "Plates, Batches, and Barcodes",
      "description": "Slides for 'Plates, Batches, and Barcodes' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Understanding Barcodes</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/mtekman/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/mtekman" alt="Mehmet Tekman">Mehmet Tekman</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What are barcodes?</p>
</li>
            
            <li><p>What is their purpose?</p>
</li>
            
            <li><p>How are barcodes formatted?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Demultiplex FASTQ data via UMI-tools</p>
</li>
            
            <li><p>Understand and validate the extraction of barcodes</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Plates, Batches, and Barcodes:
                            
                                <a href="/training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 2 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/2573177">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Barcodes are small oligonucleotides that are inserted into the captured sequence at a specific point, and provide two pieces of information about the sequence:</p>

<ol>
  <li>Which cell the sequence came from</li>
  <li>Which transcript the sequence came from</li>
</ol>

<p>When the sequence is mapped against a reference genome, we can then see which gene locus it aligns to and qualitatively assert that, together with the two pieces of information above, the sequence depicts a transcript from a specific gene that originated from a specific cell.</p>

<p>Barcodes come in a variety of formats, and in this tutorial we will be looking at the <a href="https://doi.org/10.1186/s13059-016-0938-8">CEL-Seq2 protocol</a> used in droplet-based <span class="notranslate">single-cell</span> <span class="notranslate">RNA</span>-seq.</p>

<h3 id="the-cel-seq2-protocol">The CEL-Seq2 Protocol</h3>

<p><small><a href="javascript:window.history.back()">Back to previous</a></small></p>

<p>CEL-Seq2 is a paired-end protocol, meaning that two primers bind to opposite ends of a c<span class="notranslate">DNA</span> strand in order to sequence. Each primer has a specific role.</p>

<p>In this case; <em>Read1</em> contains the barcoding information followed by the polyT tail of the messenger <span class="notranslate">RNA</span>, and <em>Read2</em> contains the actual sequence. Here, Read1 is regarded as the ‚Äòforward‚Äô strand and Read2 as the ‚Äòreverse‚Äô strand, though this is more a convention when dealing with paired-end data rather than an indication of the actual strand orientation.</p>

<figure id="figure-1"><img src="../../images/celseq2_schema.svg" alt="CEL-Seq2 Scheme" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Read1 encapsulates the barcodes, and Read2 the m<span class="notranslate">RNA</span> sequence</figcaption></figure>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#understanding-barcodes" id="markdown-toc-understanding-barcodes">Understanding Barcodes</a></li>
  <li><a href="#mitigating-duplicate-transcript-counts-with-umis" id="markdown-toc-mitigating-duplicate-transcript-counts-with-umis">Mitigating duplicate transcript counts with UMIs</a>    <ol>
      <li><a href="#naive-amplification" id="markdown-toc-naive-amplification">Naive Amplification</a></li>
      <li><a href="#amplification-with-umis" id="markdown-toc-amplification-with-umis">Amplification with UMIs</a></li>
    </ol>
  </li>
  <li><a href="#barcoding-format" id="markdown-toc-barcoding-format">Barcoding Format</a>    <ol>
      <li><a href="#our-four-reads-of-interest" id="markdown-toc-our-four-reads-of-interest">Our Four <span class="notranslate">Reads</span> of Interest</a></li>
      <li><a href="#verifying-the-barcode-format" id="markdown-toc-verifying-the-barcode-format">Verifying the Barcode Format</a></li>
    </ol>
  </li>
  <li><a href="#uniting-barcodes-with-sequence" id="markdown-toc-uniting-barcodes-with-sequence">Uniting Barcodes with Sequence</a>    <ol>
      <li><a href="#coupling-our-data-sources" id="markdown-toc-coupling-our-data-sources">Coupling our Data Sources</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="understanding-barcodes">Understanding Barcodes</h1>

<blockquote class="comment">
  <h3 id="comment-viewing-the-pre-requisite-topic"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Viewing the pre-requisite topic</h3>

  <p>It is highly recommended that the material in the <strong><em>Plates, Batches, and Barcodes</em></strong> slides <a href="../scrna-plates-batches-barcodes/slides.html">here</a> are first observed before proceeding with the rest of this tutorial.</p>

</blockquote>

<p>Cell barcodes are designed primarily for delineating one cell from another, such that read transcripts containing different cell barcodes can be trivially said to be derived from different cells.</p>

<figure id="figure-2"><img src="../../images/scrna_pbb_barcodes_add.svg" alt="Cell Barcodes" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Two seperate cell barcodes GGG and TCT, added to all read transcripts of two different cells</figcaption></figure>

<p>Transcript barcodes, meanwhile, are random sets of nucleotides added to each transcript.</p>

<figure id="figure-3"><img src="../../images/scrna_umi_add.svg" alt="Transcript Barcodes" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Random oligonucleotides barcodes added to (cell barcoded) <span class="notranslate">reads</span></figcaption></figure>

<p>There are two things to take note of:</p>

<ol>
  <li>The number of duplicates in the transcript barcodes (left)</li>
  <li>The number of duplicate read transcripts (right)</li>
</ol>

<p>Transcript barcodes are often <em>not</em> unique. This becomes evident when you consider that there are approximately 200,000 m<span class="notranslate">RNA</span>‚Äôs in a given mammalian cell (<a href="https://doi.org/10.1038/nrg3542">ref</a>) which would require barcode lengths of greater than 9 nucleotides to capture, assuming no sequencing errors.</p>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Why is it important to know which cell a read came from?</li>
    <li>Why do we need to barcode a read transcript too? Isn‚Äôt <span class="notranslate">mapping</span> it against the reference genome enough?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>If all our <span class="notranslate">reads</span> encode for a <em>Red Gene</em> (as above), we may want to know which cells express <em>Red Gene</em> more than others.
        <ul>
          <li>e.g. If our <em>Grey</em> cell has 10 times more <em>Red Gene</em> <span class="notranslate">reads</span> than our <em>Green</em> cell, then we know that the <em>Grey</em> cell and <em>Green</em> cell differ in their <span class="notranslate">expression</span> of <em>Red Gene</em> - which might be biologically significant.</li>
        </ul>
      </li>
      <li><em>Yes</em> and <em>no</em>!
        <ul>
          <li><strong>Yes</strong>: We can indeed align our sequence against a reference genome and obtain the name of the gene it aligns against. This sequence will then contribute to the ‚Äòcount‚Äô of sequences that gene has, and increase the <span class="notranslate">expression</span> of that gene.</li>
          <li><strong>No</strong>: We do not know whether these ‚Äòcounts‚Äô are <em>unique</em>. Many of these counts could be duplicates as a result of the amplification process. To explain further, we must look at UMIs and their role in the analysis.</li>
        </ul>
      </li>
    </ol>
  </blockquote>

</blockquote>

<p>The purpose of transcript barcodes is to reduce the impact of duplicated <span class="notranslate">reads</span> that occur non-linearly during the amplification process.</p>

<p>For this reason, transcript barcodes do not need to be unique. As long as we know that a given read maps to a specific transcript (i.e. after <span class="notranslate">mapping</span> it to a transcriptome), then we can assess how unique that read is based on:</p>

<ol>
  <li>Cell barcode</li>
  <li>Transcript barcode</li>
  <li>Mapped location</li>
</ol>

<p>To fully explore the uniqueness of counts, we must discuss the inclusion of <em>UMIs</em> in a <span class="notranslate">single-cell</span> analysis.</p>

<h1 id="mitigating-duplicate-transcript-counts-with-umis">Mitigating duplicate transcript counts with UMIs</h1>

<p>One of the major issues with sequencing is that the read fragments require amplification before they can be sequenced. A gene with a single m<span class="notranslate">RNA</span> transcript will not be detected by most sequencers, so it needs to be duplicated 100-1000x times for the sequencer to ‚Äòsee‚Äô it.</p>

<p>Amplification is an imprecise process however, since some <span class="notranslate">reads</span> are amplified more than others, and subsequent amplification can lead to these over-amplified <span class="notranslate">reads</span> being over-amplified even more, leading to an exponential bias of some <span class="notranslate">reads</span> over others.</p>

<p><small><strong><em>Note:</em></strong> Cell barcodes are not shown in any of the below examples, we assume they were added to our transcripts previously. </small><br /><br /></p>

<h2 id="naive-amplification">Naive Amplification</h2>

<figure id="figure-4"><img src="../../images/scrna_amplif_errors.svg" alt="Amplification Bias" /><figcaption><span class="figcaption-prefix">Figure 4:</span> A cell with two <span class="notranslate">reads</span> from different transcripts being amplified unevenly</figcaption></figure>

<p>Consider the above example where two <span class="notranslate">reads</span> from different transcripts are amplified unevenly. The resulting frequency table would yield:</p>

<blockquote class="matrix">
  <table>
    <thead>
      <tr>
        <th>¬†</th>
        <th><span class="notranslate">Reads</span> in Cell 1</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Gene Red</td>
        <td>4</td>
      </tr>
      <tr>
        <td>Gene Blue</td>
        <td>0</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<p>But the truth is entirely different (i.e. Gene Red should have 1 count, and Gene Blue should also have 1 count).
How do we correct for this bias?</p>

<h2 id="amplification-with-umis">Amplification with UMIs</h2>

<p><strong>Unique Molecular Identifiers</strong> (or <em>UMIs</em>) constitute the second portion of a barcode, where their role is to <em>uniquely</em> count <span class="notranslate">reads</span> such that amplicons of the same read are only counted once, e.g:</p>

<figure id="figure-5"><img src="../../images/scrna_amplif_errors_umis.svg" alt="Amplification Bias with UMIs" /><figcaption><span class="figcaption-prefix">Figure 5:</span> A cell with four unique transcripts, two from Gene Red and two from Gene Blue</figcaption></figure>

<p>Here, we see two unique transcripts from Gene Red and two unique transcripts from Gene Blue, each given a (coloured) UMI. After amplification, Gene Red has more <span class="notranslate">reads</span> than Gene Blue. If we were to construct a frequency table as before to count the <span class="notranslate">reads</span>, we would have:</p>

<blockquote class="matrix">
  <table>
    <thead>
      <tr>
        <th>¬†</th>
        <th><span class="notranslate">Reads</span> in Cell 1</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Gene Red</td>
        <td>6</td>
      </tr>
      <tr>
        <td>Gene Blue</td>
        <td>3</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<p>This information is false, because it shows that Red has twice the <span class="notranslate">expression</span> that Blue does. However, we can reconstitute the true count by considering the UMI information:</p>

<blockquote class="matrix">
  <table>
    <thead>
      <tr>
        <th>¬†</th>
        <th>UMI colour</th>
        <th><span class="notranslate">Reads</span> in Cell 1</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Gene Red</td>
        <td>Pink</td>
        <td>2</td>
      </tr>
      <tr>
        <td>¬†</td>
        <td>Blue</td>
        <td>4</td>
      </tr>
      <tr>
        <td>Gene Blue</td>
        <td>Brown</td>
        <td>1</td>
      </tr>
      <tr>
        <td>¬†</td>
        <td>Green</td>
        <td>2</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<p>From this we can then make the decision to ignore the frequencies of these UMIs, and simply count how many <em>unique</em> UMIs we see in each gene:</p>

<blockquote class="matrix">
  <table>
    <thead>
      <tr>
        <th>¬†</th>
        <th>Set of UMIs in Gene</th>
        <th>UMIs in Cell 1</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Gene Red</td>
        <td>{Pink, Blue}</td>
        <td>2</td>
      </tr>
      <tr>
        <td>Gene Blue</td>
        <td>{Brown, Green}</td>
        <td>2</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<p>This then provides us with the true count of the number of true transcripts for each gene as given by our original figure.</p>

<blockquote class="matrix">
  <table>
    <thead>
      <tr>
        <th>¬†</th>
        <th>UMIs in Cell 1</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Gene Red</td>
        <td>2</td>
      </tr>
      <tr>
        <td>Gene Blue</td>
        <td>2</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-about-umis"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions about UMIs</h3>

  <ol>
    <li>Are UMIs not specific to certain genes? Can the same UMI map to different genes?</li>
    <li>Can the same UMI map to different m<span class="notranslate">RNA</span> molecules of the same gene?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Yes, UMIs are not specific to genes and the same UMI barcode can tag the transcripts of different genes. UMIs are not universal tags, they are just ‚Äòadded randomness‚Äô that help reduce amplification bias.</li>
      <li>Yes, UMIs are not precise but operate probabilistically. In most cases, two transcripts of the same gene will be tagged by different UMIs. In rarer (but still prevalent) cases, the same UMI will capture different transcripts of the same gene.
        <ul>
          <li>One helpful way to think about how quantification is performed is to observe the following hierarchy of data <code class="language-plaintext highlighter-rouge">Cell Barcode ‚Üí Gene ‚Üí UMI</code></li>
        </ul>
      </li>
    </ol>

    <p>e.g.</p>

    <table>
      <thead>
        <tr>
          <th>BC:Cell</th>
          <th>Maps to Gene</th>
          <th>BC:UMI</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>AAAT</td>
          <td>Slx1</td>
          <td>TCA</td>
        </tr>
        <tr>
          <td>AAAT</td>
          <td>Slx2</td>
          <td>GTG</td>
        </tr>
        <tr>
          <td>AAAT</td>
          <td>Gh13</td>
          <td>TCA</td>
        </tr>
        <tr>
          <td>TTAA</td>
          <td>Slx1</td>
          <td>TCA</td>
        </tr>
        <tr>
          <td>TTAA</td>
          <td>Atp3</td>
          <td>CCC</td>
        </tr>
      </tbody>
    </table>

    <p>If UMIs were unique to a gene, then the <code class="language-plaintext highlighter-rouge">TCA</code> UMI barcode would not have <span class="notranslate">reads</span> that map to both <em>Slx1</em> and <em>Gh13</em> in the same cell (<code class="language-plaintext highlighter-rouge">AAAT</code>).</p>

  </blockquote>

</blockquote>

<h1 id="barcoding-format">Barcoding Format</h1>

<p>We now know the role of UMIs and cell barcodes, but how do we handle them in the analysis? Let us look at 4 example sequences in our paired-end FASTQ data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-preparing-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Preparing the Data</h3>

  <ol>
    <li>Create a new history and rename it (<em>e.g.</em> ‚ÄòInspecting FastQ Files in sc<span class="notranslate">RNA</span> batch data‚Äô)</li>
    <li>Import the following files from <a href="https://zenodo.org/record/2573177"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2573177/files/test_barcodes_celseq2_R1.fastq.gz
https://zenodo.org/record/2573177/files/test_barcodes_celseq2_R2.fastq.gz
</code></pre></div>      </div>
    </li>
  </ol>

  <blockquote class="tip">

    <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

    <ul>
      <li>Copy the link location</li>
      <li>
        <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
      </li>
      <li>Select <strong>Paste/Fetch Data</strong></li>
      <li>
        <p>Paste the link into the text field</p>
      </li>
      <li>
        <p>Press <strong>Start</strong></p>
      </li>
      <li><strong>Close</strong> the window</li>
    </ul>

    <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

  </blockquote>
  <p><br /></p>

  <ol>
    <li>Build a <em>Dataset pair</em> for the two FASTQ files <!-- cannot force this to count from 3 -->
      <ul>
        <li>Click the <em>Operations on multiple datasets</em> check box at the top of the history panel</li>
        <li>Check the two boxes next to the R1 and R2 sc<span class="notranslate">RNA</span> FASTQ samples</li>
        <li>Click <em>For all selected‚Ä¶</em> and choose <em>Build dataset pair</em></li>
        <li>Ensure that the forward read is the <code class="language-plaintext highlighter-rouge">R1</code> sample, and the reverse read is the <code class="language-plaintext highlighter-rouge">R2</code> sample.
          <ul>
            <li>Click ‚ÄòSwap‚Äô otherwise.</li>
          </ul>
        </li>
        <li>Set the name of the pair</li>
      </ul>
    </li>
    <li>Generate a list of <span class="notranslate">reads</span> to filter by creating a plain tabular file containing the following read names:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>J00182:75:HTKJNBBXX:2:1114:12469:11073
J00182:75:HTKJNBBXX:2:2222:13301:35690
J00182:75:HTKJNBBXX:2:1203:25022:13763
J00182:75:HTKJNBBXX:2:1115:8501:46961
</code></pre></div>      </div>
    </li>
    <li>Set the datatype of the file as <strong>tabular</strong></li>
  </ol>

</blockquote>

<p>At this point we now have a history with two items: our paired FASTQ test data, and a tabular file of read names. We will now apply the tabular file to the FASTQ file and extract only those <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extracting-the-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extracting the <span class="notranslate">Reads</span></h3>

  <ol>
    <li>Extracting our 4 <span class="notranslate">reads</span>
      <ul>
        <li><strong>Filter sequences by ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
          <ul>
            <li><strong>Sequence file to be filtered</strong>
              <ul>
                <li>Click the <em>Dataset Collection</em> icon</li>
                <li>Select the FastQ collection if not already selected.</li>
              </ul>
            </li>
            <li><strong>Filter using the ID list from</strong>:<code class="language-plaintext highlighter-rouge">tabular file</code>
              <ul>
                <li><em>Tabular file containing sequence identifiers</em>:<code class="language-plaintext highlighter-rouge">Pasted Entry</code></li>
              </ul>
            </li>
            <li><strong>Column(s) containing sequence identifiers</strong>
              <ul>
                <li><strong>Select/Unselect all</strong>:(tick the box)</li>
              </ul>
            </li>
            <li><strong>Output positive matches, negative matches, or both?</strong>:<code class="language-plaintext highlighter-rouge">Just positive matches (ID on list), as a single file</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Change the datatypes of the output pair to <code class="language-plaintext highlighter-rouge">fastqsanger</code> if not already set.
      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select ``</li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>Viewing our 4 <span class="notranslate">reads</span> side-by-side
      <ul>
        <li>Activate the <strong>Scratchbook</strong> by clicking on the <strong>Enable/Disable Scratchbook</strong> icon on the main top toolbar</li>
        <li>Click on the newly generated FastQ pair ending in <em>‚Äúwith matched ID‚Äù</em> to expand the individual <span class="notranslate">reads</span>
          <ul>
            <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol of the forward read</li>
            <li>Click somewhere outside the white box to close the <strong>Scratchbook</strong></li>
            <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol of the reverse read</li>
          </ul>
        </li>
        <li>Position/Resize the boxes as desired</li>
      </ul>
    </li>
  </ol>
</blockquote>

<!--
These are reads that all map to ENSDARG00000019692. In [Cell, UMI] format:

(J00182:75:HTKJNBBXX:2:1114:12469:11073|J00182:75:HTKJNBBXX:2:2222:13301:35690|J00182:75:HTKJNBBXX:2:1203:25022:13763|J00182:75:HTKJNBBXX:2:1115:8501:46961)

            Cell  , UMI
1: 46961 -- ACCAGA, GGAAGA
2: 13763 -- GGTAAC, GTCCCA -> same UMI, same cell
3: 35690 -- GGTAAC, GTCCCA -> same UMI, same cell
4: 11073 -- GGTAAC, CGGCGT -> diff UMI, same cell
-->

<h2 id="our-four-reads-of-interest">Our Four <span class="notranslate">Reads</span> of Interest</h2>

<p>Let us examine these four <span class="notranslate">reads</span> of interest which we have just sub-selected using their headers:</p>

<blockquote class="details">
  <h3 id="details-forward-reads"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Forward <span class="notranslate">Reads</span>:</h3>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@J00182:75:HTKJNBBXX:2:1115:8501:46961 1:N:0:ATCACG
GGAAGAACCAGATTTTTTTTTTTTTTTTTT
+
AAFFFJJJJJJJFFFJJJJJJJJJJJJJJJ

@J00182:75:HTKJNBBXX:2:1203:25022:13763 1:N:0:ATCACG
GTCCCAGGTAACTTTTTTTTTTTTTTTTTT
+
AAFFFJJJJJJJJFFJJJJJJJJJFJ&lt;FF-

@J00182:75:HTKJNBBXX:2:2222:13301:35690 1:N:0:ATCACG
GTCCCAGGTAACTTTTTTTTTTTTTTTTTT
+
AAFFFJJJJJJJ&lt;AFJJJJJFFJJFJJJFF

@J00182:75:HTKJNBBXX:2:1114:12469:11073 1:N:0:ATCACG
CGGCGTGGTAACTTTTTTTTTTTTTTTTCC
+
AAFFFJJJJJJJFAFFJJJJJJJJF---&lt;F
</code></pre></div>  </div>

</blockquote>

<blockquote class="details">
  <h3 id="details-reverse-reads"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Reverse <span class="notranslate">Reads</span>:</h3>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@J00182:75:HTKJNBBXX:2:1115:8501:46961 2:N:0:ATCACG
GACCTCTGATCTTTACGAAAGGCCAACGCGTTTTCAGTCTGGACACGGTTCAGCTCCTGTTCATTATTCA
+
A&lt;&lt;A-777F&lt;AA&lt;AJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ

@J00182:75:HTKJNBBXX:2:1203:25022:13763 2:N:0:ATCACG
GCCACCTAATTTCCGTCATCACACTCCTCTCCGTTTTCAACTTGCACAATGCTGTCTCCGCAGAATCCCT
+
---&lt;----&lt;A---77-7A-FJ&lt;JJFFJJ&lt;JJAJ7&lt;-FAFFJJFF&lt;FFJJFFAJFA-AFFFJFFFFFJAJJ

@J00182:75:HTKJNBBXX:2:2222:13301:35690 2:N:0:ATCACG
CAATCCTCTCCGTTATCAACTTGCACAATGCTGTCTCCGCAGAATCCCTCCGGATCAGGATCGCTCTCCA
+
&lt;&lt;A-77--77F&lt;----7AFJ-A--FJJJFAJF-AFAJAJ&lt;JFJ&lt;JJJFFJJJFJJJJJAAFJJJFJJJF-

@J00182:75:HTKJNBBXX:2:1114:12469:11073 2:N:0:ATCACG
ATCCACTTATTGCAAAGCAGAGGACATTGAGTCTCACCTTTTGTCCAGGTCTTCCAATTTCACCCTGCAA
+
A-77AA-7FF&lt;7FFJFFFJJJJJJJJJJJJJ-AFJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
</code></pre></div>  </div>

</blockquote>

<p>What we observe are the standard four lines of any FASTQ file:</p>
<ol>
  <li>Read name starting with <code class="language-plaintext highlighter-rouge">@</code></li>
  <li>Sequence of nucleotide bases</li>
  <li>Seperator <code class="language-plaintext highlighter-rouge">+</code></li>
  <li>Quality string of the nucleotide bases in ASCII</li>
</ol>

<p>The main source of interest for us is in the (2) sequences of these <span class="notranslate">reads</span>, which somewhere within encode for three crucial pieces of information that we will need to perform quantification:</p>

<ol>
  <li>Cell Barcode</li>
  <li>UMI Barcode</li>
  <li>Reverse-transcribed m<span class="notranslate">RNA</span> sequence</li>
</ol>

<p>These can be encoded into the sequences of our paired-end data by any means. In order to know where our barcodes are, we must be familiar with the sequencing primers used in the analysis:</p>

<!-- TODO: Expand the schema to talk about the actual read primers and not just the sequencing primers -->

<h2 id="verifying-the-barcode-format">Verifying the Barcode Format</h2>

<p>As shown in <a href="#the-cel-seq2-protocol">CEL-Seq2 protocol</a>, we have the following encoding:</p>
<ul>
  <li>Forward Read:
    <ul>
      <li>01-06bp: UMI Barcode</li>
      <li>07-12bp: Cell Barcode</li>
      <li>13-30bp: Poly-T tail</li>
    </ul>
  </li>
  <li>Reverse Read:
    <ul>
      <li>01-70bp: m<span class="notranslate">RNA</span> sequence</li>
    </ul>
  </li>
</ul>

<p>The encoding of the barcodes on the first read can actually be seen by examining the distribution of bases in a FastQC plot.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-confirming-the-barcoding"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Confirming the Barcoding</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúShort read data from your current history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired FastQ</code> (the original paired set) You will need to choose ‚ÄòDataset collection‚Äô to allow this as an input.</li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>We are only interested in the distribution of bases on the Forward read, but it is more convenient to process the data as a pair instead of un-hiding the original dataset</p>
      </blockquote>
    </li>
    <li>Click on <em>FastQC on collection :Webpage</em></li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> of the Forward read</li>
    <li>Click on the <em>Per base sequence content</em> header on the side-bar</li>
  </ol>
</blockquote>

<figure id="figure-6"><img src="../../images/scrna_barcodes_fastq.png" alt="FastQC Plot" /><figcaption><span class="figcaption-prefix">Figure 6:</span> FastQC plot of Forward read</figcaption></figure>

<p>Here we can see the three distinct regions along the x-axis that correspond to our expected <a href="#the-cel-seq2-protocol">CEL-Seq2 Schema</a>:</p>

<ul>
  <li>01-06bp: smooth, relatively constant bases.</li>
  <li>07-12bp: noisy, highly varied distribution of bases.</li>
  <li>13-30bp: T-dominated region</li>
</ul>

<p>We can see that the distribution of nucleotides in the 01-06bp range is relatively more stable than the distribution of nucleotides in the 07-12bp range, which seems to exhibit more extreme variation.</p>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Why is this the case? Why is the UMI barcode base distribution smoother than the Cell barcode base distribution?</p>
  <blockquote class="solution">

    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>There are far more UMIs than cells. Cell barcodes are designed and selected with a specified edit distance, greatly limiting their availability in the data. UMIs are not so well-curated ‚Äì i.e it is possible to encounter the same UMI in the same cell multiple times. The more extreme variation in the 7-12bp region is simply caused by a fewer number of samples.</p>

  </blockquote>
</blockquote>

<h1 id="uniting-barcodes-with-sequence">Uniting Barcodes with Sequence</h1>

<p>In a sense, we have a disparity in our data: the reverse <span class="notranslate">reads</span> contain the sequences we wish to map, but not the barcodes; the forward <span class="notranslate">reads</span> contain the barcode, but not the sequence. For the forward and reverse <span class="notranslate">reads</span> given above, the information that we really want from both can be summarized in this table:</p>

<table>
  <thead>
    <tr>
      <th>Read</th>
      <th>Cell</th>
      <th>UMI</th>
      <th>Sequence</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><small>@J00182:75:HTKJNBBXX:2:1115:8501:46961</small></td>
      <td>ACCAGA</td>
      <td>GGAAGA</td>
      <td><small>GACCTCTGATCTTTACGAAAGGCCAACGCGTTTTCAGTCTGGACACGGTTCAGCTCCTGTTCATTATTCA</small></td>
    </tr>
    <tr>
      <td><small>@J00182:75:HTKJNBBXX:2:1203:25022:13763</small></td>
      <td>GGTAAC</td>
      <td>GTCCCA</td>
      <td><small>GCCACCTAATTTCCGTCATCACACTCCTCTCCGTTTTCAACTTGCACAATGCTGTCTCCGCAGAATCCCT</small></td>
    </tr>
    <tr>
      <td><small>@J00182:75:HTKJNBBXX:2:2222:13301:35690</small></td>
      <td>GGTAAC</td>
      <td>GTCCCA</td>
      <td><small>CAATCCTCTCCGTTATCAACTTGCACAATGCTGTCTCCGCAGAATCCCTCCGGATCAGGATCGCTCTCCA</small></td>
    </tr>
    <tr>
      <td><small>@J00182:75:HTKJNBBXX:2:1114:12469:11073</small></td>
      <td>GGTAAC</td>
      <td>CGGCGT</td>
      <td><small>ATCCACTTATTGCAAAGCAGAGGACATTGAGTCTCACCTTTTGTCCAGGTCTTCCAATTTCACCCTGCAA</small></td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Provided that these <span class="notranslate">reads</span> all map to the same gene:</p>

  <ol>
    <li>Which of these <span class="notranslate">reads</span> come from the same cell?</li>
    <li>Which of these <span class="notranslate">reads</span> are PCR duplicates?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li><span class="notranslate">Reads</span>:
        <ul>
          <li><code class="language-plaintext highlighter-rouge">@J00182:75:HTKJNBBXX:2:1203:25022:13763</code></li>
          <li><code class="language-plaintext highlighter-rouge">@J00182:75:HTKJNBBXX:2:2222:13301:35690</code></li>
          <li><code class="language-plaintext highlighter-rouge">@J00182:75:HTKJNBBXX:2:1114:12469:11073</code>
 all have the cell barcode <code class="language-plaintext highlighter-rouge">GGTAAC</code>.</li>
        </ul>
      </li>
      <li><span class="notranslate">Reads</span>:
        <ul>
          <li><code class="language-plaintext highlighter-rouge">@J00182:75:HTKJNBBXX:2:1203:25022:13763</code></li>
          <li><code class="language-plaintext highlighter-rouge">@J00182:75:HTKJNBBXX:2:2222:13301:35690</code>
 appear to be PCR duplicates, since they both have the same cell barcode and same UMI.</li>
        </ul>
      </li>
    </ol>

    <p>However if we consider their sequences, we can see that they contain different (but overlapping) sequences.</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>13763:   GCCACCTAATTTCCGTCATCACACTCCTCTCCGTTTTCAACTTGCACAATGCTGTCTCCGCAGAATCCCT
35690:                        CAATCCTCTCCGTTATCAACTTGCACAATGCTGTCTCCGCAGAATCCCTCCGGATCAGGATCGCTCTCCA
</code></pre></div>    </div>

    <p>They describe the same transcript but have sequences from different <span class="notranslate">reads</span>, and therefore both <span class="notranslate">reads</span> <em>should</em> be counted as seperate <span class="notranslate">reads</span>. Whether or not both these <span class="notranslate">reads</span> are counted as a single read due to their identical barcodes, or counted seperately due to their differing sequences depends entirely on the deduplication utility they are fed into it.</p>

  </blockquote>
</blockquote>

<h2 id="coupling-our-data-sources">Coupling our Data Sources</h2>

<p>How should we unite these two source of information into a single location without impacting the data content?</p>

<p>For this we need to take the barcode information from the Forward <span class="notranslate">reads</span>, and stick it into the <em>header</em> of the Reverse <span class="notranslate">reads</span>. That way we can align our sequence to the reference and still keep the barcode information associated with the <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-barcode-extraction-and-annotation-of-our-4-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Barcode Extraction and Annotation of our 4 <span class="notranslate">reads</span></h3>

  <ol>
    <li><strong>UMI-tools extract</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúLibrary type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Our paired set of 4 sequences)</li>
            <li><em>‚ÄúBarcode on both <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcode on first read only</code></li>
          </ul>
        </li>
        <li><em>‚ÄúUse Known Barcodes?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúBarcode pattern for first read‚Äù</em>: <code class="language-plaintext highlighter-rouge">NNNNNNCCCCCC</code></li>
        <li><em>‚ÄúEnable quality filter?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <ul>
          <li>Here we specify the format of our barcodes as <code class="language-plaintext highlighter-rouge">NNNNNNCCCCCC</code> where the <em>N</em>s represent UMI bases and the <em>C</em>s represent the cell barcodes.</li>
          <li>In some protocols, actual sequence data can be found in between the cell and UMI barcodes, wherein it is neccesary to represent sequence bases using <em>X</em>.</li>
          <li>e.g. A protocol that starts with a 3bp sequence, followed by a 4bp Cell barcode, followed once again by a 10bp sequence, and then finally a 5bp UMI barcode, would require the following barcode format:
<code class="language-plaintext highlighter-rouge">XXXCCCCXXXXXXXXXXNNNNN</code></li>
        </ul>

      </blockquote>
    </li>
    <li>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol on the <em><span class="notranslate">Reads</span>1: UMI-tools extract</em> file</li>
    <li>Click somewhere outside the white box to close the <strong>Scratchbook</strong></li>
    <li>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol on the <em><span class="notranslate">Reads</span>2: UMI-tools extract</em> file</li>
  </ol>

</blockquote>

<p>We should now be able to see the following <span class="notranslate">reads</span>:</p>

<blockquote class="details">
  <h3 id="details-forward-reads-1"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Forward <span class="notranslate">Reads</span>:</h3>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@J00182:75:HTKJNBBXX:2:1115:8501:46961_ACCAGA_GGAAGA 1:N:0:ATCACG
TTTTTTTTTTTTTTTTTT
+
FFFJJJJJJJJJJJJJJJ

@J00182:75:HTKJNBBXX:2:1203:25022:13763_GGTAAC_GTCCCA 1:N:0:ATCACG
TTTTTTTTTTTTTTTTTT
+
JFFJJJJJJJJJFJ&lt;FF-

@J00182:75:HTKJNBBXX:2:2222:13301:35690_GGTAAC_GTCCCA 1:N:0:ATCACG
TTTTTTTTTTTTTTTTTT
+
&lt;AFJJJJJFFJJFJJJFF

@J00182:75:HTKJNBBXX:2:1114:12469:11073_GGTAAC_CGGCGT 1:N:0:ATCACG
TTTTTTTTTTTTTTTTCC
+
FAFFJJJJJJJJF---&lt;F&gt;
</code></pre></div>  </div>

</blockquote>

<blockquote class="details">
  <h3 id="details-reverse-reads-1"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Reverse <span class="notranslate">Reads</span>:</h3>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@J00182:75:HTKJNBBXX:2:1115:8501:46961_ACCAGA_GGAAGA 2:N:0:ATCACG
GACCTCTGATCTTTACGAAAGGCCAACGCGTTTTCAGTCTGGACACGGTTCAGCTCCTGTTCATTATTCA
+
A&lt;&lt;A-777F&lt;AA&lt;AJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ

@J00182:75:HTKJNBBXX:2:1203:25022:13763_GGTAAC_GTCCCA 2:N:0:ATCACG
GCCACCTAATTTCCGTCATCACACTCCTCTCCGTTTTCAACTTGCACAATGCTGTCTCCGCAGAATCCCT
+
---&lt;----&lt;A---77-7A-FJ&lt;JJFFJJ&lt;JJAJ7&lt;-FAFFJJFF&lt;FFJJFFAJFA-AFFFJFFFFFJAJJ

@J00182:75:HTKJNBBXX:2:2222:13301:35690_GGTAAC_GTCCCA 2:N:0:ATCACG
CAATCCTCTCCGTTATCAACTTGCACAATGCTGTCTCCGCAGAATCCCTCCGGATCAGGATCGCTCTCCA
+
&lt;&lt;A-77--77F&lt;----7AFJ-A--FJJJFAJF-AFAJAJ&lt;JFJ&lt;JJJFFJJJFJJJJJAAFJJJFJJJF-

@J00182:75:HTKJNBBXX:2:1114:12469:11073_GGTAAC_CGGCGT 2:N:0:ATCACG
ATCCACTTATTGCAAAGCAGAGGACATTGAGTCTCACCTTTTGTCCAGGTCTTCCAATTTCACCCTGCAA
+
A-77AA-7FF&lt;7FFJFFFJJJJJJJJJJJJJ-AFJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
</code></pre></div>  </div>

</blockquote>

<p>Notice the remaining sequence in each of the <span class="notranslate">reads</span>, and that the reverse <span class="notranslate">reads</span> appear to fully encapsulate all the information that we wanted to capture in our table at the beginning of this section.</p>

<blockquote class="question">
  <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Compare the Forward/Read1 and Reverse/Read2 <span class="notranslate">reads</span> to those prior the extraction. What has happened to the header and sequence of each read?</li>
    <li>Are the Forward <span class="notranslate">reads</span> useful at all?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Comparison:
        <ul>
          <li>Forward:
            <ul>
              <li>Sequence: The <code class="language-plaintext highlighter-rouge">cell</code> and <code class="language-plaintext highlighter-rouge">umi</code> sections of the sequence have been removed, leaving behind only the PolyT tail.</li>
              <li>Header: The <code class="language-plaintext highlighter-rouge">cell</code> and <code class="language-plaintext highlighter-rouge">umi</code> sections of the sequence have been added as <code class="language-plaintext highlighter-rouge">cell_umi</code> barcode in the header</li>
            </ul>
          </li>
          <li>Reverse:
            <ul>
              <li>Sequence: Has not changed.</li>
              <li>Header: The <code class="language-plaintext highlighter-rouge">cell</code> and <code class="language-plaintext highlighter-rouge">umi</code> sections of the sequence from the Forward (note: <strong>NOT</strong> Reverse) <span class="notranslate">reads</span> have been added to the header.</li>
            </ul>
          </li>
        </ul>
      </li>
      <li>With the inclusion of the cell and UMI barcodes into the header of our sequence data, we now have all our useful data in the Reverse <span class="notranslate">reads</span>. We can now effectively throw away our Forward <span class="notranslate">reads</span>, as they have no more useful information within them.</li>
    </ol>
  </blockquote>
</blockquote>

<p>We have now successfully multiplexed data from several different (cell) sources into a single file that will greatly simplify the <span class="notranslate">mapping</span>/alignment process <span class="notranslate">downstream</span>.</p>

<p>We have also now successfully <em>de</em>-multiplexed our data, by decoding each pair of <span class="notranslate">reads</span> into barcoding and sequence parts and making use of the FASTQ format by storing this information within the FASTQ headers and data, respectively.</p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>With this tutorial we have understood the importance of handling FASTQ data from different sources, and extracting the information we need (barcodes (cell and UMI) and sequence) using <strong>UMI-tools</strong> so that we can perform <span class="notranslate">mapping</span> without losing any context of where the <span class="notranslate">reads</span> are derived from.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Verifying the distribution of barcodes via a FASTQC plot</p>
</li>
                            
                            <li><p>Relocating barcodes into headers</p>
</li>
                            
                            <li><p>Removing unwanted barcodes</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Understanding Barcodes (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Mehmet Tekman, 2021 <b>Understanding Barcodes (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html">/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-scrna-umis,
    author = "Mehmet Tekman",
    title = "Understanding Barcodes (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Pre-processing of Single-Cell RNA Data:
                            
                                <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Mehmet Tekman)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/scrna-umis/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
