---
layout: tutorial_slides
logo: "GTN"
title: "Plates, Batches, and Barcodes"
zenodo_link: ""
tags:
  - single-cell
questions:
  - "What are barcodes and how are they applied to batches?"
  - "What is the difference between a plate and a batch?"
  - "Are batches and lanes the same?"
  - "Why are plating setups important to know?"
  - "Is it neccesary to check barcodes that don't exist in a batch?"
objectives:
  - "How to describe a plating setup"
  - "Proper naming conventions when dealing with multiple batches"
  - "Checking for false positives across batches"
time_estimation: "1h"
key_points:
  - "Plates are split into lanes"
  - "Batches are lanes, but their names are not reused"
  - "Multiple batches can exist in many Plates"
  - "Checking all barcodes across batches eliminates false positives"
requirements:
  - 
    type: "internal"
    topic_name: sequence-analysis
    tutorials:
        - scrna-introduction

follow_up_training:
  - 
    type: "internal"
    topic_name: sequence-analysis
    tutorials:
        - scrna_preprocessing
        
contributors:
  - mtekman

---

### Sequencing Plates


                      | · · · · · · · · · |  ← Plate: N=9, M=6 
                      | · · · · · · · · · |           54 slots 
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     

.left[Plates are *N x M* arrays of wells that cells are sorted, to then be individually amplified and sequenced.]

---

### Sequencing Plates

                      | o o · · o o · · · |  ← Plate: N=9, M=6 
                      | o o · · o o · · · |           54 slots 
                      | o o · · o o · · · |                      
                      | o o · · o o · · · |     24 cells        
                      | o o · · o o · · · |     30 empty slots  
                      | o o · · o o · · · |                      

.left[Not all wells need to be occupied by cells, some wells are left empty.]
.center[Why is this the case?]

---

### Sequencing Plates

                      | o o % % o o % · · |  ← Plate: N=9, M=6 
                      | o o % % o o % · · |           54 slots 
                      | o o % % o o % · · |                      
                      | o o % % o o % · · |     24 cells         
                      | o o % % o o % · · |     30 empty slots   
                      | o o % % o o % · · |                      

.center[Library preparation is not always perfect]
.center[Sample **contamination** from neighbouring wells]

---

### Plates and Lanes


        | · · · · · · · · · |  <- Plate: N=9, M=6
        | · · · · · · · · · |            54 slots
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |

.center[Sequencing Plates are divided into *lanes*:]


        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |
        
---

### Plates and Lanes


        | · · · · · · · · · |  <- Plate: N=9, M=6
        | · · · · · · · · · |            54 slots
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |

.center[Sequencing Plates are divided into *lanes*:]


        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |


* Lanes demarcate evenly-sized *n x m* rectangular regions on a plate (where *n < N and m < M*).
* Multiple lanes can exist on plate, lanes ideally do not overlap within a plate.

---

### Plates and Lanes

        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |

* All slots within a lane are sequenced at the same time.
* All lanes within a plate *may or may not* be sequenced at the same time.
* Lanes can therefore be thought of as different *batches*.

---

### Lanes and Batches

           L1       L2     L3                                  L1      L2      L3
        | · · · : · · · : · · · |  <---   Plate 1          | · · · : · · · : · · · |
        | · · · : · · · : · · · |       with 3 lanes       | · · · : · · · : · · · |
        | · · · : · · · : · · · |                          | · · · : · · · : · · · |
        | · · · : · · · : · · · |         Plate 2   --->   | · · · : · · · : · · · |
        | · · · : · · · : · · · |       with 3 lanes       | · · · : · · · : · · · |
        | · · · : · · · : · · · |                          | · · · : · · · : · · · |
           B1       B2     B3                                  B4      B5      B6

* Lanes and batches describe the same spaces, but not the same samples.
* For this reason, batch numbers should *not* persist across different plates.

---

### Distinguishing cells in a plate

                          L1       L2     L3      L4
      Plate 1 -->     | · · · : · · · : · · · : · · · |
    with 4 lanes      | · · · : · · · : · · · : · · · |
    and 12 slots      | · · · : · · · : · · · : · · · |
      per lane        | · · · : · · · : · · · : · · · |
                          B1       B2     B3      B4     

* Cells are selected from a plate by their *barcodes*
* Barcodes must be unique
  + e.g. *4 x 12  = 48* slots in the plate, need 48 unique cell barcodes

???
The way these slots are filled depends entirely on the protocol, but usually not all slots are filled. The reason for this will become clear momentarily

---

### What are Cell Barcodes?

.image-75[![Add Barcodes](../../images/scrna_pbb_barcodes_add.svg)]

* Barcodes are added to *all* transcripts of a *specific* cell
* Transcripts with different cell barcodes originate from different cells

???


---

### What are Cell Barcodes?

.image-100[![Barcodes in Plate](../../images/scrna_pbb_barcodes_overview.svg)]

* Many different cell barcodes are used across many different cells
* Each well in a plate contains a cell, indexed by its cell barcode

---

### Questions about Cell Barcodes

.left[Assuming a lane is a 20x5 array:]

1. How many cell barcodes are needed for a single lane?

1. How many cell barcodes are needed for a plate with 10 lanes?

1. What would be the minimum length of the barcodes for each of the previous questions?
--
.center[.footnote[
1. *20 x 5 = 100 unique barcodes per lane*
1. *20 x 5 x 10 = 1,000 unique barcodes per plate*
]]
---

### Questions about Cell Barcodes

.left[3) What would be the minimum length of the barcodes for each of the previous questions?]

.pull-left[

A single lane?

 * 100 barcodes, 4 nucleotides for each base of a barcode

| Barcodes | Result |
|---------|------|
| $$4^2 =  16$$ | <small>No, 2 bases is not enough |
| $$4^3  = 64$$ | <small>No, 3 bases is not enough |
| $$4^4 = 256$$ | <small>Yes, 4 bases is enough to cover 100 barcodes (and more!) |

]
--
.pull-right[
  
A plate with 10 lanes?

 * 1000 barcodes, 4 nucleotides for each base of a barcode

| Barcodes | Result |
|---------|------|
|$$4^4 = 256$$ | <small>No, 4 bases is not enough |
|$$4^5 = 1024$$ | <small>Yes, 5 bases is enough to cover 1,000 barcodes (just barely!) |

]

---

### Barcode Safeguarding

<!-- TODO:
 * Find a way to get <style>.MathJax_Display { display: inline; }</style> working so that MathJax can render inline
 * The MathJax selector seems to apply directly to the element, which is impossible to override!
 -->

.left[* Is 5 nucleotides really enough to capture 1000 cells?]
.left[* If we assume our cell barcodes look like:]


    AAAAA AAAAC AAAAG AAAAT AAACA AAAGA AAATA ....
    CCCCC CCCCA CCCCG CCCCT CCCAC CCCGC CCCTC ....
      .
      .
      .


.left[* What could go wrong if all barcodes are seperated by 1 bp, as above?]

???
If we assume that every barcode is seperated from every other barcode by 1 bp, then the answer is 'yes'

That's right, sequencing errors can mislabel a read to a completely different cell than from where the transcript originated from.

---

### Guarding against Sequencing Errors

.left[* Sequencing errors can change the cell barcode]
.left[* To guard against a 1 bp sequencing error, need to space barcodes by 2 bp]
.left[* With an *Edit Distance* of 2 bp:]


    AAAAA AAACC AAAGG AAATT AACCA AAGGA AATTA ...
    CCCCC CCCAA CCCGG CCCTT CCAAC CCGGC CCTTC ...
      .
      .
      .

.left[* How many barcodes would this yield for a cell barcode of length 5 bp?]

$$ $$
--
$$ 4^{5-1} = 512 $$


---

### Edit Distance : General Principle

.pull-left[
e.g. For barcodes of length N=3:

* Edit Distance of E=1:  
.center[`AAA AAC AAG AAT ACA ACC` ...]
.center[<small>64 barcodes</small>]

* Edit Distance of E=2:  
.center[`AAA ACC AGG ATT CAA CCC` ...] 
.center[<small>16 barcodes</small>]

* Edit Distance of E=3:  
.center[`AAA CCC GGG TTT`]
.center[<small>4 barcodes</small>]
]
--
.pull-right[
Number of Barcodes :

$$ 4^{N-(E-1)} $$

.center[For barcodes of length *N*,  
and Edit Distance of *E*.]

]

---

### Design Factors in Barcodes

Is it better to use Longer or Shorter Barcodes?

.pull-left[
Longer:
* Offer larger range of barcodes
* Prone to more sequencing errors
   * Must increase edit distance significantly
* Can accomodate large edit distances
]

.pull-right[
Shorter:
* Offer smaller range of barcodes 
* More robust against sequencing errors
  * A smaller edit distance is more acceptable
* *Cannot* accomodate large edit distances
]

---

### Cell Barcodes: Summary

.pull-left[
* A single barcode sequence indexes a single cell
* Every transcript in a specific cell has the same cell barcode
]
.pull-right[
* Barcodes are *designed*
* Barcodes are *limited*
*  
*  
]
--
.center[
#### Need to balance:  
* Edit Distance vs. Barcode Length
* Number of Barcodes vs. Plate and Lane Size
  * ***How?***
]

---

### Intelligent Plating Strategies

The main contending questions are:

 * How large is each batch?
 
 * How many batches on a plate?
 
 * Should each batch use the same barcodes?
 
 * What constraints are there on the plate?
 

 
???
A technician always has to balance quality against cost, and this is illustrated in the following examples:

---

# Example Setup

.pull-left[
* Barcodes:
  * 24 unique barcodes, with an edit distance of E=2:
         
             AAA ACC AGG TTT TAA TCC
             ATT CCC CAA TGG NAA NCC
             CGG CTT GGG NGG NTT ANN
             GAA GCC GTT CNN GNN TNN

]
--
.pull-right[
* Plates and Lanes:
  * 12 slots per lane
  * 4 lanes per plate
  
* Contraints
  * Only 2 lanes sequenced at a time
    * i.e. half the plate is sequenced at a time
]

???
Here we use N as an extra base just for example purposes, but you do sometimes see this in other barcodes.
2 lanes at a time, only half

---

### Example 1: Single Plate with a Single Lane

.footnote[
<small>
Available Barcodes

         
                                           AAA ACC AGG TTT TAA TCC
                                           ATT CCC CAA TGG NAA NCC
                                           CGG CTT GGG NGG NTT ANN
                                           GAA GCC GTT CNN GNN TNN

</small>
]

* 12 total slots in Plate 1
* 12 slots in Lane 1
* All slots filled

           | 1 1 1 |    <- AAA ACC AGT     
           | 1 1 1 |       ATT CCC CAA      
           | 1 1 1 |       CGG CTT GGG      
           | 1 1 1 |       GAA GCC GTT

--
* We only need to use half of our barcodes

--

* Why is this wasteful?


???
* Half of the barcodes used for that lane, and the other half we can ignore.
* Wasteful because we are not getting full use of our 24 barcodes in a single sequecing run

---

### Example 2: Single Plate with 2 Lanes

.footnote[
<small>
Available Barcodes

         
                                           AAA ACC AGG TTT TAA TCC
                                           ATT CCC CAA TGG NAA NCC
                                           CGG CTT GGG NGG NTT ANN
                                           GAA GCC GTT CNN GNN TNN

</small>
]

* 24 total slots in Plate 1
* 12 slots in Lane 1, and 12 slots in Lane 2
* All slots filled

         | 1 1 1 : 2 2 2 | <- AAA ACC AGG : TTT TAA TCC
         | 1 1 1 : 2 2 2 |    ATT CCC CAA : TGG NAA NCC
         | 1 1 1 : 2 2 2 |    CGG CTT GGG : NGG NTT ANN
         | 1 1 1 : 2 2 2 |    GAA GCC GTT : CNN GNN TNN

--
* We can use all of our barcodes
* Maximum number of cells can be sequenced in a single run

--
* Why might this be too optimal?

???
Here we use all barcodes since these lanes will be sequenced at the same time 
Let's look at one final example to see why using all our barcodes on a plate might not be optimal.

---

### Example 3: Single Plate with 2 Lanes, only 1 active

<style>
.bottom-right-box {
  position: absolute;
  bottom:0px;
  border: 1px solid grey;
  font-size: medium;
  padding-left: 10px;
  padding-right: 10px;
  margin-left:auto;
  margin-right:auto;
  width: 50%;
}
</style>


.bottom-right-box[
Available Barcodes
        
    AAA ACC AGG TTT TAA TCC
    ATT CCC CAA TGG NAA NCC
    CGG CTT GGG NGG NTT ANN
    GAA GCC GTT CNN GNN TNN

]

* 24 total slots in Plate 1
* 12 slots in Lane 1, and 12 slots in Lane 2
* Only slots in Lane 1 filled
 
         | 1 1 1 : . . . | <- AAA ACC AGG : ttt taa tcc
         | 1 1 1 : . . . |    ATT CCC CAA : tgg naa ncc
         | 1 1 1 : . . . |    CGG CTT GGG : ngg ntt ann
         | 1 1 1 : . . . |    GAA GCC GTT : cnn gnn tnn

--
* Why have we not filled in Lane 2?



???
All barcodes used

---





# Questions about Cell Barcodes

1. Why is it important to know which cell a sequence came from?

2. Barcoding the cell makes sense, but why do we need to barcode all transcripts of a cell too? 

???
Can we not infer which gene the sequence originates from by simply mapping it against the reference genome?

---

# Questions about Barcodes

1. Cell Barcodes: Why is it important to know which cell a sequence came from?
  * *GeneX* may exist is many cells. The relative expression of a gene between two different cells may be of biological significance.
    
2. Cell Barcodes make sense, but why do we need to barcode all transcripts of a cell too?
  * **Yes** and **No**!
  * **Yes**:  We can count 
 
 ???
 * If our sequence codes for a *GeneX* which is a gene of interest, we may want to know which cells express GeneX more than others. 
      * e.g. If *CellA* has 10 times more *GeneX* sequences than *CellB*, then we know that *CellA* and *CellB* differ at *GeneX* - which might suggest a causative source of variation for any change in function between *CellA* and *CellB* (or cells in the same cluster as *CellA*, and cells in the same cluster as *CellB*).

 * *Yes* and *no*!  
    * **Yes**: We can indeed align our sequence against a reference genome and obtain the name of the gene it aligns against. This sequence will then contribute to the 'count' of sequences that gene has, and increase the expression of that gene.    
    * **No**: We do not know whether these 'counts' are *unique*. Many of these counts could be duplicates as a result of the amplification process. To explain further, we must look at UMIs and their role in the analysis.


???
Can we not infer which gene the sequence originates from by simply mapping it against the reference genome?

---
