---
layout: tutorial_slides
logo: "GTN"
title: "Plates, Batches, and Barcodes"
zenodo_link: ""
tags:
  - single-cell
questions:
  - "What are barcodes and how are they applied to batches?"
  - "What is the difference between a plate and a batch?"
  - "Are batches and lanes the same?"
  - "Why are plating setups important to know?"
  - "Is it neccesary to check barcodes that don't exist in a batch?"
objectives:
  - "How to describe a plating setup"
  - "Proper naming conventions when dealing with multiple batches"
  - "Checking for false positives across batches"
time_estimation: "1h"
key_points:
  - "Plates are split into lanes"
  - "Batches are lanes, but their names are not reused"
  - "Multiple batches can exist in many Plates"
  - "Checking all barcodes across batches eliminates false positives"
requirements:
  - 
    type: "internal"
    topic_name: sequence-analysis
    tutorials:
        - scrna-introduction

follow_up_training:
  - 
    type: "internal"
    topic_name: sequence-analysis
    tutorials:
        - scrna_preprocessing
        
contributors:
  - mtekman

---

# Sequencing Plates

                      | · · · · · · · · · |  ← Plate: N=9, M=6 
                      | · · · · · · · · · |           54 slots 
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     
                      | · · · · · · · · · |                     

.left[Plates are *N x M* arrays of wells that cells are sorted, to then be individually amplified and sequenced.]

---

# Sequencing Plates

                      | o o · · o o · · · |  ← Plate: N=9, M=6 
                      | o o · · o o · · · |           54 slots 
                      | o o · · o o · · · |                      
                      | o o · · o o · · · |     24 cells        
                      | o o · · o o · · · |     30 empty slots  
                      | o o · · o o · · · |                      

.left[Not all wells need to be occupied by cells, some wells are left empty.]
.center[Why is this the case?]

---

# Sequencing Plates

                      | o o % % o o % · · |  ← Plate: N=9, M=6 
                      | o o % % o o % · · |           54 slots 
                      | o o % % o o % · · |                      
                      | o o % % o o % · · |     24 cells         
                      | o o % % o o % · · |     30 empty slots   
                      | o o % % o o % · · |                      

.center[Library preparation is not always perfect]
.center[Sample **contamination** from neighbouring wells]

---

# Plates and Lanes


        | · · · · · · · · · |  <- Plate: N=9, M=6
        | · · · · · · · · · |            54 slots
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |

.center[Sequencing Plates are divided into *lanes*:]


        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |
        
---

# Plates and Lanes


        | · · · · · · · · · |  <- Plate: N=9, M=6
        | · · · · · · · · · |            54 slots
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |                    
        | · · · · · · · · · |

.center[Sequencing Plates are divided into *lanes*:]


        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |


* Lanes demarcate evenly-sized *n x m* rectangular regions on a plate (where *n < N and m < M*).
* Multiple lanes can exist on plate, lanes ideally do not overlap within a plate.

---

# Plates and Lanes

        | · · · : · · · : · · · |  <---   Plate: N=9, M=6, 3 lanes
        | · · · : · · · : · · · |                54 slots
        | · · · : · · · : · · · |          Lane: n=3, m=6
        | · · · : · · · : · · · |                18 slots per lane
        | · · · : · · · : · · · |
        | · · · : · · · : · · · |

* All slots within a lane are sequenced at the same time.
* All lanes within a plate *may or may not* be sequenced at the same time.
* Lanes can therefore be thought of as different *batches*.

---

# Lanes and Batches

           L1       L2     L3                                  L1      L2      L3
        | · · · : · · · : · · · |  <---   Plate 1          | · · · : · · · : · · · |
        | · · · : · · · : · · · |       with 3 lanes       | · · · : · · · : · · · |
        | · · · : · · · : · · · |                          | · · · : · · · : · · · |
        | · · · : · · · : · · · |         Plate 2   --->   | · · · : · · · : · · · |
        | · · · : · · · : · · · |       with 3 lanes       | · · · : · · · : · · · |
        | · · · : · · · : · · · |                          | · · · : · · · : · · · |
           B1       B2     B3                                  B4      B5      B6

* Lanes and batches describe the same spaces, but not the same samples.
* For this reason, batch numbers should *not* persist across different plates.

---

# Distinguishing cells in a plate

                          L1       L2     L3      L4
      Plate 1 -->     | · · · : · · · : · · · : · · · |
    with 4 lanes      | · · · : · · · : · · · : · · · |
    and 12 slots      | · · · : · · · : · · · : · · · |
      per lane        | · · · : · · · : · · · : · · · |
                          B1       B2     B3      B4     

* Cells are selected from a plate by their *barcodes*
* Barcodes must be unique
  + e.g. *4 x 12  = 48* slots in the plate, need 48 unique cell barcodes

???
The way these slots are filled depends entirely on the protocol, but usually not all slots are filled. The reason for this will become clear momentarily

---

# What are Cell Barcodes?

.image-75[![Add Barcodes](../../images/scrna_pbb_barcodes_add.svg)]

* Barcodes are added to *all* transcripts of a *specific* cell
* Transcripts with different cell barcodes originate from different cells

???


---

# What are Cell Barcodes?

.image-100[![Barcodes in Plate](../../images/scrna_pbb_barcodes_overview.svg)]

* Many different cell barcodes are used across many different cells
* Each well in a plate contains a cell, indexed by its cell barcode

---

# Questions about Cell Barcodes

.left[Assuming a lane is a 20x5 array:]

1. How many cell barcodes are needed for a single lane?

1. How many cell barcodes are needed for a plate with 10 lanes?

1. What would be the minimum length of the barcodes for each of the previous questions?


---

# Questions about Cell Barcodes

.left[Assuming a lane is a 20x5 array:]

1. How many cell barcodes are needed for a single lane?
    * 20 x 5 = 100 unique barcodes per lane

1. How many cell barcodes are needed for a plate with 10 lanes?
    * 20 x 5 x 10 = 1,000 unique barcodes per plate
    
---

# Questions about Cell Barcodes?

3) What would be the minimum length of the barcodes for each of the previous questions?
  
.pull-left[

1)   A single lane?

 * 100 barcodes, 4 nucleotides for each base of a barcode

| 4^2 = 16 | No, 2 bases is not enough |
| 4^3  = 64 | No, 3 bases is not enough |
| 4^4 = 256 | Yes, 4 bases is enough to cover 100 barcodes (and more!) |
]

.pull-right[
  
  2.  A plate with 10 lanes?
     - 1000 barcodes, 4 nucleotides for each base of a barcode
     - 4^4 = 256 ? No, 4 bases is not enough
     - 4^5 = 1024 ? Yes, 5 bases is enough to cover 1,000 barcodes (just barely!)

---















---

# Questions about Cell Barcodes

1. Why is it important to know which cell a sequence came from?

2. Barcoding the cell makes sense, but why do we need to barcode all transcripts of a cell too? 

???
Can we not infer which gene the sequence originates from by simply mapping it against the reference genome?

---

# Questions about Barcodes

1. Cell Barcodes: Why is it important to know which cell a sequence came from?
  * *GeneX* may exist is many cells. The relative expression of a gene between two different cells may be of biological significance.
    
2. Cell Barcodes make sense, but why do we need to barcode all transcripts of a cell too?
  * **Yes** and **No**!
  * **Yes**:  We can count 
 
 ???
 * If our sequence codes for a *GeneX* which is a gene of interest, we may want to know which cells express GeneX more than others. 
      * e.g. If *CellA* has 10 times more *GeneX* sequences than *CellB*, then we know that *CellA* and *CellB* differ at *GeneX* - which might suggest a causative source of variation for any change in function between *CellA* and *CellB* (or cells in the same cluster as *CellA*, and cells in the same cluster as *CellB*).

 * *Yes* and *no*!  
    * **Yes**: We can indeed align our sequence against a reference genome and obtain the name of the gene it aligns against. This sequence will then contribute to the 'count' of sequences that gene has, and increase the expression of that gene.    
    * **No**: We do not know whether these 'counts' are *unique*. Many of these counts could be duplicates as a result of the amplification process. To explain further, we must look at UMIs and their role in the analysis.


???
Can we not infer which gene the sequence originates from by simply mapping it against the reference genome?

---
