<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Object tracking using CellProfiler</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Image analysis using Galaxy" />
        <meta property="og:title" content="Galaxy Training: Object tracking using CellProfiler" />
        <meta property="og:description" content="Image analysis using Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/imaging" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Imaging
                        </a>
                        
                    </li>
                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/imaging/faqs/" title="Check our FAQs for the Imaging topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Translate this page
            </div>

            <div id="lang-selector">
                
                <strong>Automatic Translations</strong>

<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url">
                Fran√ßais
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url">
                Êó•Êú¨Ë™û
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url">
                Espa√±ol
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url">
                Portugu√™s
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url">
                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                </a>
                
                <a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html&edit-text=&act=url" title="">
                And more!
                </a>
            </div>
        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-10-01/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html" title="Version 2021-10-01">2021-10-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-09-01/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html" title="Version 2021-09-01">2021-09-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-08-01/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html" title="Version 2021-08-01">2021-08-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Object tracking using CellProfiler",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Imaging",
    "description": "Image analysis using Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/imaging/"
  },
  "courseCode": "imaging / object-tracking-using-cell-profiler / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Object tracking using CellProfiler' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - How to segment and track objects in fluorescence time-lapse microscopy images?\n\n\\nThe objectives are:\n - Segment fluorescent objects using CellProfiler in Galaxy\n - Track objects over multiple frames using CellProfiler in Galaxy\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html",
      "name": "Creating, Editing and Importing Galaxy Workflows",
      "description": "Hands-on for 'Creating, Editing and Importing Galaxy Workflows' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Yi Sun"
    },
    {
      "@type": "Person",
      "name": "Beatriz Serrano-Solano"
    },
    {
      "@type": "Person",
      "name": "Jean-Karim H√©rich√©"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Yi Sun"
    },
    {
      "@type": "Person",
      "name": "Beatriz Serrano-Solano"
    },
    {
      "@type": "Person",
      "name": "Jean-Karim H√©rich√©"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Imaging",
      "description": "Image analysis using Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/imaging/"
    }
  ]
}
</script>

<section class="tutorial topic-imaging">
    <h1 data-toc-skip>Object tracking using CellProfiler</h1>
    

    <div class="contributors-line">Authors: <a href="/training-material/hall-of-fame/sunyi000/" class="contributor-badge contributor-sunyi000"><img src="https://avatars.githubusercontent.com/sunyi000?s=27" alt="Avatar">Yi Sun</a><a href="/training-material/hall-of-fame/beatrizserrano/" class="contributor-badge contributor-beatrizserrano"><img src="https://avatars.githubusercontent.com/beatrizserrano?s=27" alt="Avatar">Beatriz Serrano-Solano</a><a href="/training-material/hall-of-fame/jkh1/" class="contributor-badge contributor-jkh1"><img src="https://avatars.githubusercontent.com/jkh1?s=27" alt="Avatar">Jean-Karim H√©rich√©</a></div>

    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to segment and track objects in fluorescence time-lapse microscopy images?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Segment fluorescent objects using CellProfiler in Galaxy</p>
</li>
        
        <li><p>Track objects over multiple frames using CellProfiler in Galaxy</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/galaxy-interface">Using Galaxy and Managing your Data</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Creating, Editing and Importing Galaxy Workflows:
                            
                            
                                
                                     <a href="/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" href="http://doi.org/10.5281/zenodo.4567084">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/workflows/" title="Workflows" alt="Object tracking using CellProfiler workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/imaging/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 23, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Most biological processes are dynamic and observing them over time can provide valuable insights. Combining fluorescent markers with time-lapse imaging is a common approach to collect data on dynamic cellular processes such as cell division (e.g. <span class="citation"><a href="#Neumann2010">Neumann <i>et al.</i> 2010</a></span>, <span class="citation"><a href="#Heriche2014">H√©rich√© <i>et al.</i> 2014</a></span>). However, automated time-lapse imaging can produce large amounts of data that can be challenging to process. One of these challenges is the tracking of individual objects as it is often impossible to manually follow a large number of objects over many time points.</p>

<p>To demonstrate how automatic tracking can be applied in such situations, this tutorial will track dividing nuclei in a short time-lapse recording of one mitosis of a syncytial blastoderm stage <em>Drosophila</em> embryo expressing a GFP-histone gene that labels chromatin.</p>

<p>Tracking is done by first segmenting objects then linking objects between consecutive frames. Linking is done by matching objects and several criteria or matching rules are available. Here we will link objects if they significantly overlap between the current and previous frames.</p>

<blockquote class="tip">
  <h3 id="tip-prerequisites"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Prerequisites</h3>
  <p>It is expected that you are already familiar with the <span class="notranslate">Galaxy</span> interface and the <span class="notranslate">workflow</span> editor. If this is not the case, we recommend you to complete the requirements listed at the start of this tutorial.</p>
</blockquote>

<blockquote class="warning">
  <h3 id="warning-important-information-cellprofiler-in-galaxy"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> <strong>Important information: CellProfiler in <span class="notranslate">Galaxy</span></strong></h3>
  <p>The <span class="notranslate">Galaxy</span> <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0"><strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+<span class="notranslate">galaxy</span>0</span></span>   tool takes two inputs: a CellProfiler pipeline and an image collection.
Some pipelines created with stand-alone CellProfiler may not work with the <span class="notranslate">Galaxy</span> CellProfiler tool. Some reasons are:</p>
  <ul>
    <li>The pipeline was built with a different version of CellProfiler. The <span class="notranslate">Galaxy</span> tool currently uses CellProfiler 3.9.</li>
    <li>Modules used by the pipeline aren‚Äôt available in <span class="notranslate">Galaxy</span>.</li>
    <li>Parameters for some CellProfiler modules are limited/constrained compared to the stand-alone version, most notably:
      <ul>
        <li>Parameters require manual input from the user whereas, in the stand-alone version, some modules can inherit parameter values from other modules.</li>
        <li>Input and output file locations are set by <span class="notranslate">Galaxy</span> and can‚Äôt be set by the user.</li>
        <li>Metadata extraction from file names is limited to a set of fixed patterns.</li>
      </ul>
    </li>
  </ul>

  <p>It is recommended to build a CellProfiler pipeline using the <span class="notranslate">Galaxy</span> interface if the pipeline is to be run by <span class="notranslate">Galaxy</span>.</p>
</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#create-a-cellprofiler-pipeline-in-galaxy" id="markdown-toc-create-a-cellprofiler-pipeline-in-galaxy">Create a CellProfiler pipeline in <span class="notranslate">Galaxy</span></a>    <ol>
      <li><a href="#create-a-new-workflow" id="markdown-toc-create-a-new-workflow">Create a new <span class="notranslate">workflow</span></a></li>
      <li><a href="#read-the-images" id="markdown-toc-read-the-images">Read the images</a></li>
      <li><a href="#get-the-metadata" id="markdown-toc-get-the-metadata">Get the metadata</a></li>
      <li><a href="#convert-the-images-to-grayscale" id="markdown-toc-convert-the-images-to-grayscale">Convert the images to grayscale</a></li>
      <li><a href="#segmentation" id="markdown-toc-segmentation">Segmentation</a></li>
      <li><a href="#feature-extraction" id="markdown-toc-feature-extraction">Feature extraction</a></li>
      <li><a href="#track-nuclei" id="markdown-toc-track-nuclei">Track nuclei</a></li>
      <li><a href="#visualize-results" id="markdown-toc-visualize-results">Visualize results</a></li>
      <li><a href="#save-the-images-and-features" id="markdown-toc-save-the-images-and-features">Save the images and features</a></li>
    </ol>
  </li>
  <li><a href="#run-the-pipeline-with-cellprofiler" id="markdown-toc-run-the-pipeline-with-cellprofiler">Run the pipeline with <strong>CellProfiler</strong></a></li>
</ol>

</blockquote>

<h1 id="get-data">Get data</h1>

<p>This tutorial will use a time-lapse recording of nuclei progressing through mitotic anaphase during early <em>Drosophila</em> embryogenesis. The nuclei are labelled on chromatin with a GFP-histone marker and have been imaged every 7 seconds using a laser scanning confocal microscope with a 40X objective.
The images are saved as a zip archive on Zenodo and need to be uploaded to the <span class="notranslate">Galaxy</span> server before they can be used.</p>

<figure id="figure-1"><img src="../../images/object-tracking-using-cell-profiler/Dividing_nuclei.gif" alt="Dividing nuclei. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Time lapse recording of anaphase nuclei in a Drosophila embryo.</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial.
When you log in for the first time, an empty, unnamed history is created by default. You can simply rename it.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
    </li>
    <li>Import <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> the files from <a href="http://doi.org/10.5281/zenodo.4567084">Zenodo</a> or from
the shared data library.
      <ul>
        <li><strong>Important:</strong> If setting the type to ‚ÄòAuto-detect‚Äô, make sure that after upload, the datatype is set to zip.</li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/api/files/e5d1bd5c-60a0-42e4-8f0d-a2ebc863c5d9/drosophila_sample.zip
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li>    <p><strong>Close</strong> the window</p>  </li>  <li>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</li></ul></blockquote>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> the file to drosophila_embryo.zip</li>
  </ol>

</blockquote>

<h1 id="create-a-cellprofiler-pipeline-in-galaxy">Create a CellProfiler pipeline in <span class="notranslate">Galaxy</span></h1>

<p>In this section, we will build a CellProfiler pipeline from scratch in <span class="notranslate">Galaxy</span>.
We need to:</p>
<ul>
  <li>Read the images and the metadata</li>
  <li>Convert the colour images to grayscale</li>
  <li>Segment the nuclei</li>
  <li>Extract features from the segmented nuclei</li>
  <li>Perform tracking</li>
  <li>Produce some useful output files</li>
</ul>

<p>A pipeline is built by chaining together <span class="notranslate">Galaxy</span> tools representing CellProfiler modules and must start with the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1"><strong>Starting modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+<span class="notranslate">galaxy</span>1</span></span>  tool and end with the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0"><strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+<span class="notranslate">galaxy</span>0</span></span>   tool.</p>

<figure id="figure-2"><img src="../../images/object-tracking-using-cell-profiler/CP_object_tracking_pipeline.png" alt="Image of the workflow. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Overview of the CellProfiler pipeline using <span class="notranslate">Galaxy</span> tools.</figcaption></figure>

<blockquote class="details">
  <h3 id="details-more-details-about-the-pipeline-steps"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More details about the pipeline steps</h3>
  <ul>
    <li>Metadata is needed to tell CellProfiler what a temporal sequence of images is and what the order of images is in the sequence.</li>
    <li>CellProfiler is designed to work primarily with grayscale images. Since we don‚Äôt need the colour information, we convert colour images to grayscale type.</li>
    <li>Segmentation means identifying the nuclei in each image. In CellProfiler, this is done by thresholding the intensity level in each image.</li>
    <li>When we perform tracking we‚Äôre usually interested in quantifying how some properties of the objects evolve over time. Also, sometimes we may want to do tracking by matching objects based on some property of the objects (e.g. a shape measurement). Therefore, for each segmented object, we compute some features, i.e. numerical descriptors of some properties of the object.</li>
    <li>Tracking will provide the information required to allow <span class="notranslate">downstream</span> data analysis tools to link the features into a multidimensional time series.</li>
  </ul>

</blockquote>

<h2 id="create-a-new-workflow">Create a new <span class="notranslate">workflow</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-new-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a new <span class="notranslate">workflow</span></h3>
  <ol>
    <li>
      <p>Create a new <span class="notranslate">workflow</span></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new <span class="notranslate">workflow</span></h3>  <ol>  <li>Click <strong><span class="notranslate">Workflow</span></strong> on the top bar</li>  <li>Click the new <span class="notranslate">workflow</span> <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-new</span> button</li>  <li>Give it a clear and memorable name</li>  <li>Clicking <strong>Save</strong> will take you directly into the <span class="notranslate">workflow</span> editor for that <span class="notranslate">workflow</span></li>  <li>Need more help? Please see the <em>How to make a <span class="notranslate">workflow</span> subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li></ol></blockquote>
    </li>
  </ol>

</blockquote>

<p>The next steps will add new tools using the <span class="notranslate">workflow</span> editor.
Remember to save the <span class="notranslate">workflow</span> when done (or anytime) to not lose your input parameters.</p>

<h2 id="read-the-images">Read the images</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-reading-images"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Reading images</h3>

  <ol>
    <li>
      <p>Add Inputs ‚Üí <strong>Input Dataset</strong> to the <span class="notranslate">workflow</span></p>
    </li>
    <li>
      <p>Add <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Unzip</strong> with the following parameter:</p>
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúExtract single file‚Äù</em>: <code class="language-plaintext highlighter-rouge">All files</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="get-the-metadata">Get the metadata</h2>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>The Starting modules tool combines four CellProfiler modules that are always used together at the start of a pipeline. These modules are:</p>
  <ul>
    <li>Images</li>
    <li>Metadata</li>
    <li>NamesAndTypes</li>
    <li>Groups</li>
  </ul>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-getting-metadata"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Getting metadata</h3>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+galaxy1"><strong>Starting modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_common/cp_common/3.1.9+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:</p>

      <ul>
        <li>Images
          <ul>
            <li><em>‚ÄúDo you want to filter only the images?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Select the images only</code></li>
          </ul>
        </li>
        <li>Metadata
          <ul>
            <li>
              <p><em>‚ÄúDo you want to extract the metadata?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, specify metadata</code></p>

              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Insert new metadata
                  <ul>
                    <li><em>‚ÄúMetadata extraction method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Extract from file/folder names</code></li>
                    <li><em>‚ÄúMetadata source‚Äù</em>: <code class="language-plaintext highlighter-rouge">File name</code></li>
                    <li><em>‚ÄúSelect the pattern to extract metadata from the file name‚Äù</em>: <code class="language-plaintext highlighter-rouge">field1_field2_field3</code></li>
                    <li>
                      <p><em>‚ÄúExtract metadata from‚Äù</em>: <code class="language-plaintext highlighter-rouge">Images matching a rule</code></p>

                      <ul>
                        <li><em>‚ÄúMatch the following rules‚Äù</em>: <code class="language-plaintext highlighter-rouge">All</code></li>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Insert filtering rules:
                          <ul>
                            <li><em>‚ÄúSelect the filtering criteria‚Äù</em>: <code class="language-plaintext highlighter-rouge">File</code></li>
                            <li><em>"‚Äùoperator</em>: <code class="language-plaintext highlighter-rouge">Does</code></li>
                            <li><em>‚Äúcontain‚Äù</em>: <code class="language-plaintext highlighter-rouge">Contain regular <span class="notranslate">expression</span></code></li>
                            <li><em>‚Äúmatch_value‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFPHistone</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>NamesAndTypes
          <ul>
            <li><em>‚ÄúProcess 3D‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, do not process 3D data</code></li>
            <li><em>‚ÄúAssign a name to‚Äù</em>: <code class="language-plaintext highlighter-rouge">Give images one of several names depending on the metadata</code>
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Insert another image:
                  <ul>
                    <li><em>‚ÄúMatch the following rules‚Äù</em>: <code class="language-plaintext highlighter-rouge">All</code></li>
                    <li>
                      <p><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Insert rule:</p>

                      <ul>
                        <li>
                          <p><em>‚ÄúSelect rule criteria‚Äù</em>: <code class="language-plaintext highlighter-rouge">File</code></p>

                          <ul>
                            <li><em>‚Äúoperator‚Äù</em>: <code class="language-plaintext highlighter-rouge">Does</code></li>
                            <li><em>‚Äúcontain‚Äù</em>: <code class="language-plaintext highlighter-rouge">Contain regular <span class="notranslate">expression</span></code></li>
                            <li><em>‚Äúmatch_value‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFPHistone</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><em>‚ÄúSelect the image type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Color image</code></li>
                    <li><em>‚ÄúName to assign these images‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigColor</code></li>
                    <li><em>‚ÄúSet intensity range from‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image metadata</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúImage matching method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Metadata</code></li>
          </ul>
        </li>
        <li>Groups
          <ul>
            <li><em>‚ÄúDo you want to group your images?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, group the images</code></li>
            <li><em>‚ÄúMetadata category‚Äù</em>: <code class="language-plaintext highlighter-rouge">field1</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>How are we capturing metadata and what type of metadata are we getting?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>Metadata is obtained from the filenames by extracting three text fields separated by an underscore. The metadata we get is ‚ÄúDrosophilaEmbryo‚Äù, ‚ÄúGFPHistone‚Äù and numbers with leading zeros. These three fields represent respectively the sample identifier, the marker visualized in the image and the index in the time series.</p>

  </blockquote>

</blockquote>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>How will CellProfiler form a temporal sequence?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>Images will be grouped based on field1 which here is the sample identifier and ordered alpha-numerically (by default) which will order them by field3 (the time series index) since fields 1 and 2 are constant.</p>

  </blockquote>

</blockquote>

<h2 id="convert-the-images-to-grayscale">Convert the images to grayscale</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-colour-to-grayscale-conversion"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Colour to grayscale conversion</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_color_to_gray/cp_color_to_gray/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_color_to_gray/cp_color_to_gray/3.1.9+galaxy0"><strong>ColorToGray</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_color_to_gray/cp_color_to_gray/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: Connect output of <strong>Starting Modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to input of <span><strong>ColorToGray</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span></li>
        <li><em>‚ÄúEnter the name of the input image‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigColor</code></li>
        <li><em>‚ÄúConversion method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Combine</code>
          <ul>
            <li><em>‚ÄúName the output image‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigGray</code></li>
            <li><em>‚ÄúImage type‚Äù</em>: <code class="language-plaintext highlighter-rouge">RGB</code>
              <ul>
                <li><em>‚ÄúRelative weight of the red channel‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li><em>‚ÄúRelative weight of the green channel‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li><em>‚ÄúRelative weight of the blue channel‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="segmentation">Segmentation</h2>

<p>The first step to track nuclei starts with the identification of those objects on the images.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-nuclei-segmentation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Nuclei segmentation</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_identify_primary_objects/cp_identify_primary_objects/3.1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_identify_primary_objects/cp_identify_primary_objects/3.1.9+galaxy1"><strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_identify_primary_objects/cp_identify_primary_objects/3.1.9+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>ColorToGray</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúUse advanced settings?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, use advanced settings</code>
          <ul>
            <li><em>‚ÄúEnter the name of the input image (from NamesAndTypes)‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigGray</code></li>
            <li><em>‚ÄúEnter the name of the primary objects to be identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
            <li><em>‚ÄúTypical minimum diameter of objects, in pixel units (Min)‚Äù</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
            <li><em>‚ÄúTypical maximum diameter of objects, in pixel units (Max)‚Äù</em>: <code class="language-plaintext highlighter-rouge">9999</code></li>
            <li><em>‚ÄúDiscard objects outside diameter range?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúDiscard objects touching the border of the image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúThreshold strategy‚Äù</em>: <code class="language-plaintext highlighter-rouge">Global</code>
              <ul>
                <li><em>‚ÄúThresholding method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Otsu</code>
                  <ul>
                    <li><em>‚ÄúTwo-class or three-class thresholding?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Three classes</code>
                      <ul>
                        <li><em>‚ÄúAssign pixels in the middle intensity class to the foreground or the background?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Background</code></li>
                      </ul>
                    </li>
                    <li><em>Threshold smoothing scale‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.3488</code></li>
                    <li><em>Threshold correction factor‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>Lower bound on threshold‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.01</code></li>
                    <li><em>Upper bound on threshold‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúMethod to distinguish clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Intensity</code>
              <ul>
                <li><em>‚ÄúMethod to draw dividing lines between clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Intensity</code>
                  <ul>
                    <li><em>‚ÄúAutomatically calculate size of smoothing filter for declumping?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúAutomatically calculate minimum allowed distance between local maxima?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúSpeed up by using lower-resolution mage to find local maxima?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúFill holes in identified objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">After both thresholding and declumping</code></li>
            <li><em>‚ÄúHandling of objects if excessive number of objects identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Continue</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="feature-extraction">Feature extraction</h2>

<p>Once the objects of interest (nuclei) are identified, we extract features, i.e. numerical descriptors of object properties. We do this because we may be interested in analysing the evolution of these properties over time or want to use them in the tracking procedure to match objects over time.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-shape-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Shape features</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_measure_object_size_shape/cp_measure_object_size_shape/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_measure_object_size_shape/cp_measure_object_size_shape/3.1.9+galaxy0"><strong>MeasureObjectSizeShape</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_measure_object_size_shape/cp_measure_object_size_shape/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚Äúnew object‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the object to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúCalculate the Zernike features?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-intensity-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Intensity features</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_measure_object_intensity/cp_measure_object_intensity/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_measure_object_intensity/cp_measure_object_intensity/3.1.9+galaxy0"><strong>MeasureObjectIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_measure_object_intensity/cp_measure_object_intensity/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>MeasureObjectSizeShape</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚Äúnew image‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of an image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigGray</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚Äúnew object‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the objects to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="track-nuclei">Track nuclei</h2>

<p>With the nuclei and the relevant features measured, we are now ready to start the tracking step!</p>
<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-object-tracking"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Object tracking</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_track_objects/cp_track_objects/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_track_objects/cp_track_objects/3.1.9+galaxy0"><strong>TrackObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_track_objects/cp_track_objects/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>MeasureObjectIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúEnter the name of the objects to track‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
        <li><em>‚ÄúChoose a tracking method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Overlap</code>
          <ul>
            <li><em>‚ÄúMaximum pixel distance to consider matches‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
            <li><em>‚ÄúFilter objects by lifetime?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúSelect display option?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Color and Number</code></li>
            <li><em>‚ÄúSave color-coded image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúName the output image‚Äù</em>: <code class="language-plaintext highlighter-rouge">TrackedNuclei</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="visualize-results">Visualize results</h2>

<p>To make sure that the tracking has gone as expected, we will have a look at the original images together with the results of the segmentation step. And we will visualize them together (in tiles) for easier comparison.</p>
<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualize-segmentation-outcome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualize segmentation outcome</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_overlay_outlines/cp_overlay_outlines/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_overlay_outlines/cp_overlay_outlines/3.1.9+galaxy0"><strong>OverlayOutlines</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_overlay_outlines/cp_overlay_outlines/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>TrackObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúDisplay outlines on a blank image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><em>‚ÄúEnter the name of image on which to display outlines‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigGray</code></li>
            <li><em>‚ÄúOutline display mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Color</code>
              <ul>
                <li>In <em>‚ÄúOutline‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Outline‚Äù</em>
                      <ul>
                        <li><em>‚ÄúEnter the name of the objects to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
                        <li><em>‚ÄúSelect outline color‚Äù</em>: <span style="background-color:#ff0000;border-radius:3px;border:1px solid #000;width:12px;height:12px;margin-right:5px;">¬†¬†¬†¬†¬†</span> (red)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúName the output image‚Äù</em>: <code class="language-plaintext highlighter-rouge">OutlineImage</code></li>
        <li><em>‚ÄúHow to outline‚Äù</em>: <code class="language-plaintext highlighter-rouge">Inner</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-tiling-images"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Tiling images</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_tile/cp_tile/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_tile/cp_tile/3.1.9+galaxy0"><strong>Tile</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_tile/cp_tile/3.1.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>OverlayOutlines</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúEnter the name of an input image‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigColor</code></li>
        <li><em>‚ÄúName the output image‚Äù</em>: <code class="language-plaintext highlighter-rouge">TiledImages</code></li>
        <li><em>‚ÄúTile assembly method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Within cycles</code>
          <ul>
            <li><em>‚ÄúAutomatically calculate number of rows?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúFinal number of rows‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              </ul>
            </li>
            <li><em>‚ÄúAutomatically calculate number of columns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúImage corner to begin tiling‚Äù</em>: <code class="language-plaintext highlighter-rouge">top left</code></li>
            <li><em>‚ÄúDirection to begin tiling‚Äù</em>: <code class="language-plaintext highlighter-rouge">row</code></li>
            <li><em>‚ÄúUse meander mode?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li>In <em>‚ÄúAnother image‚Äù</em>:
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Another image‚Äù</em>
                  <ul>
                    <li><em>‚ÄúEnter the name of an additional image to tile‚Äù</em>: <code class="language-plaintext highlighter-rouge">OutlineImage</code></li>
                  </ul>
                </li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Another image‚Äù</em>
                  <ul>
                    <li><em>‚ÄúEnter the name of an additional image to tile‚Äù</em>: <code class="language-plaintext highlighter-rouge">TrackedNuclei</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="save-the-images-and-features">Save the images and features</h2>

<p>The tiled images and the features computed in previous steps are now exported to the <span class="notranslate">Galaxy</span> history ready to be analyzed.</p>
<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-save-the-images"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Save the images</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_save_images/cp_save_images/3.1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_save_images/cp_save_images/3.1.9+galaxy1"><strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_save_images/cp_save_images/3.1.9+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>Tile</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúSelect the type of image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image</code>
          <ul>
            <li><em>‚ÄúSaved the format to save the image(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">png</code></li>
          </ul>
        </li>
        <li><em>‚ÄúEnter the name of the image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">TiledImages</code></li>
        <li><em>‚ÄúSelect method for constructing file names‚Äù</em>: <code class="language-plaintext highlighter-rouge">From image filename</code>
          <ul>
            <li><em>‚ÄúEnter the image file name‚Äù</em>: <code class="language-plaintext highlighter-rouge">OrigColor</code></li>
            <li><em>‚ÄúAppend a suffix to the image file name‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúText to append to the image file name‚Äù</em>: <code class="language-plaintext highlighter-rouge">_tile</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúOverwrite existing files without warning?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúWhen to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">Every cycle</code></li>
        <li><em>‚ÄúRecord the file and path information to the saved image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What will be the file names of the output images?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The files will be named DrosophilaEmbryo_GFPHistone_0000_tile.png,
DrosophilaEmbryo_GFPHistone_0001_tile.png ‚Ä¶</p>
  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-export-tabular-data-to-character-delimited-text-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Export tabular data to character-delimited text files</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_export_to_spreadsheet/cp_export_to_spreadsheet/3.1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_export_to_spreadsheet/cp_export_to_spreadsheet/3.1.9+galaxy1"><strong>ExportToSp<span class="notranslate">reads</span>heet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_export_to_sp<span class="notranslate">reads</span>heet/cp_export_to_sp<span class="notranslate">reads</span>heet/3.1.9+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: output of <strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúSelect the column delimiter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Comma (",")</code></li>
        <li><em>‚ÄúAdd a prefix to file names?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not add prefix to the file name</code></li>
        <li><em>‚ÄúOverwrite existing files without warning‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúAdd image metadata columns to your object data file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúRepresentation of Nan/Inf‚Äù</em>: <code class="language-plaintext highlighter-rouge">NaN</code></li>
        <li><em>‚ÄúCalculate the per-image mean values for object measurements?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCalculate the per-image median values for object measurements?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCalculate the per-image standard deviation values for object measurements?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCreate a GenePattern GCT file?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>By default all measurement types are exported.
This is in contrast to the stand-alone version of CellProfiler which allows choosing which measurement types to save.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="run-the-pipeline-with-cellprofiler">Run the pipeline with <strong>CellProfiler</strong></h1>

<p>As mentioned in the introduction, the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0"><strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+<span class="notranslate">galaxy</span>0</span></span>   takes as input the pipeline that we have been assembling over this tutorial and the input data. We are now ready to finally run CellProfiler!</p>
<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-running-the-pipeline-with-cellprofiler"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Running the pipeline with CellProfiler</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+galaxy0"><strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9+<span class="notranslate">galaxy</span>0</span></span>   with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPipeline file‚Äù</em>: `output of <strong>ExportToSp<span class="notranslate">reads</span>heet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúAre the input images packed into a tar archive?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúImages‚Äù</em>: output of <strong>Unzip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúDetailed logging file?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Run the <span class="notranslate">workflow</span>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>If the pipeline fails, inspect the CellProfiler log file for clues about errors.
A common source of errors is not providing the correct information to a module such as pointing a CellProfiler module at a non-existent dataset (e.g. with a typo in the name).</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<figure id="figure-3"><img src="../../images/object-tracking-using-cell-profiler/CP_object_tracking_sample_output.png" alt="Sample output. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Sample of images produced by the pipeline.</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How many nuclei have been tracked?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>Five objects were detected but only four of these were nuclei.</p>

  </blockquote>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How could we get rid of tracks that don‚Äôt correspond to nuclei?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>Objects that are not nuclei appear to be transient so one solution would be to require the tracks to have a minimal length. This can be done using the <em>‚ÄúFilter objects by lifetime‚Äù</em> parameter of the TrackObjects module.</p>

  </blockquote>

</blockquote>
<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Where is the data needed to plot change in nuclei eccentricity over time?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The data is in the csv file called Nuclei in the CellProfiler pipeline output data set.</p>

  </blockquote>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>We‚Äôve run a CellProfiler pipeline on <span class="notranslate">Galaxy</span> to segment and track dividing nuclei in a Drosophila embryo. We‚Äôve exported images to visually inspect the outcome and saved tables of computed object features in comma-separated text files for future analysis.</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>CellProfiler in Galaxy can be used to track objects in time-lapse microscopy images</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/imaging/faqs/">FAQ page for the Imaging topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/imaging#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Neumann2010">Neumann, B., T. Walter, J.-K. H√©rich√©, J. Bulkescher, H. Erfle <i>et al.</i>, 2010 <b>Phenotypic profiling of the human genome by time-lapse microscopy reveals cell division genes</b>. Nature 464: 721‚Äì727. <a href="https://doi.org/10.1038/nature08869">10.1038/nature08869</a></li>
<li id="Heriche2014">H√©rich√©, J.-K., J. G. Lees, I. Morilla, T. Walter, B. Petrova <i>et al.</i>, 2014 <b>Integration of biological data by kernels on graph nodes allows prediction of new genes involved in mitotic chromosome condensation</b> (D. G. Drubin, Ed.). Molecular Biology of the Cell 25: 2522‚Äì2536. <a href="https://doi.org/10.1091/mbc.e13-04-0221">10.1091/mbc.e13-04-0221</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Object tracking using CellProfiler (Imaging)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Yi Sun, Beatriz Serrano-Solano, Jean-Karim H√©rich√©, 2021 <b>Object tracking using CellProfiler (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html">https://training.galaxyproject.org/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{imaging-object-tracking-using-cell-profiler,
author = "Yi Sun and Beatriz Serrano-Solano and Jean-Karim H√©rich√©",
title = "Object tracking using CellProfiler (Galaxy Training Materials)",
year = "2021",
month = "03",
day = "23"
url = "\url{https://training.galaxyproject.org/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
